source/docs/software/advanced-control/trajectories/troubleshooting.rst

```diff
@@ -0,0 +1,241 @@
+    auto rightMeas = table->GetEntry("right_measurement");
+
+    frc2::PIDController leftController(DriveConstants::kPDriveVel, 0, 0);
+    frc2::PIDController rightController(DriveConstants::kPDriveVel, 0, 0);
+    frc2::RamseteCommand ramseteCommand(
+        exampleTrajectory, [this]() { return m_drive.GetPose(); },
+        frc::RamseteController(AutoConstants::kRamseteB,
+                                AutoConstants::kRamseteZeta),
+        frc::SimpleMotorFeedforward<units::meters>(
+            DriveConstants::ks, DriveConstants::kv, DriveConstants::ka),
+        DriveConstants::kDriveKinematics,
+        [this] { return m_drive.GetWheelSpeeds(); }, leftController,
+        rightController,
+        [=](auto left, auto right) {
+            auto leftReference = leftRef;
+            auto leftMeasurement = leftMeas;
+            auto rightReference = rightRef;
+            auto rightMeasurement = rightMeas;
+
+            m_drive.TankDriveVolts(left, right);
+
+            leftMeasurement.SetDouble(m_drive.GetWheelSpeeds().left.to<double>());
+            leftReference.SetDouble(leftController.GetSetpoint());
+
+            rightMeasurement.SetDouble(m_drive.GetWheelSpeeds().right.to<double>());
+            rightReference.SetDouble(rightController.GetSetpoint());
+        },
+        {&m_drive});
+
+4. Run the robot on a variety of trajectories (curved and straight line), and check to see if the actual velocity tracks the desired velocity by looking at graphs from Network Tables.
+5. If the desired and actual are off by *a lot* then you should check if the wheel diameter and ``encoderEPR`` you used for characterization were correct. If you've verified that your units and conversions are correct, then you should try recharacterizing on the same floor that you're testing on to see if you can get better data.
+
+Verify P Gain
+^^^^^^^^^^^^^
+If you completed the previous step and the problem went away then your problem can probably be found in one of the next steps. In this step we're going to verify that your wheel P controllers are well-tuned. If you're using Java then we want to turn off Ramsete so that we can just view our PF controllers on their own.
+
+1. You must re-use all the code from the previous step that logs actual vs. desired velocity (and the code that disables Ramsete, if you're using Java), except that **the P gain must be set back to its previous nonzero value.**
+2. Run the robot again on a variety of trajectories, and check that your actual vs. desired graphs look good.
+3. If the graphs do not look good (i.e. the actual velocity is very different from the desired) then you should try tuning your P gain and rerunning your test trajectories.
+
+Check Constraints
+^^^^^^^^^^^^^^^^^
+ .. note:: Make sure that your P gain is nonzero for this step and that you still have the logging code added in the previous steps. If you're using Java then you should remove the code to disable Ramsete.
```

Comment:
Leading space is causing the rendered note to be indented.
![image](https://user-images.githubusercontent.com/19977330/73302587-daf31700-41e2-11ea-9fc7-6dfed85fdc72.png)